PVSSpringBoot/src/main/java/com/example/PVSSpringBoot/Controller.java

```diff
@@ -0,0 +1,125 @@
+import com.example.PVSSpringBoot.repositories.UsersRepo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.sql.Date;
+import java.time.LocalDateTime;
+import java.util.Map;
+
+@RestController
+public class Controller {
+
+    @Autowired
+    private UsersRepo usersRepo;
+
+    @GetMapping(path = "/hello")
+    public String helloWorld(){
+        return "hello World";
+    }
+
+    @PostMapping(path = "/signup")
+    String signUp(@RequestBody Map<String, String> body){
+        var e = usersRepo.findByEmail(body.get("email"));
+        if(!e.isEmpty()){
+            return "Email already used";
+        }
+        var now = LocalDateTime.now();
+        Date date = new Date(now.getYear()-1900, now.getMonth().getValue()-1, now.getDayOfMonth());
+        User user = User.builder().email(body.get("email")).join_date(date).is_admin(false).password(body.get("password")).user_name(body.get("user_name")).build();
+        User x = usersRepo.save(user);
+        return x.toString();
+    }
+    @GetMapping("/login")
+    public UserFront logIn(@RequestBody Map<String, String> body){
+        var entry = usersRepo.findByEmail(body.get("email"));
+        if(entry.isEmpty()){
+            return new UserFront(-1, "Email not valid", "", false);
+        }
+        User user = entry.get();
+        if(!user.getPassword().equals(body.get("password"))){
+                return new UserFront(-1, "Password is wrong", "", false);
+        }
+        return new UserFront(user.getUser_id(), user.getUser_name(), user.getEmail(), user.getIs_admin());
+    }
+    @PostMapping("/setadmin")
+    public String setAdmin(@RequestBody Map<String, Long> body){
+        var enrtyAdmin = usersRepo.findById(body.get("admin"));
+        var enrtyUser = usersRepo.findById(body.get("user"));
+        if(enrtyAdmin.isEmpty()){
+            return "Wrong admin id";
+        }
+        if(enrtyUser.isEmpty()){
+            return "Wrong user id";
+        }
+        User admin = enrtyAdmin.get();
+        User user = enrtyUser.get();
+        if(!admin.getIs_admin()){
+            return "Admin is not admin";
+        }
+        user.setIs_admin(true);
+        usersRepo.save(user);
+        return "Admined successfully";
+    }
+    @PostMapping("/removeadmin")
+    public String removeAdmin(@RequestBody Map<String, Long> body){
+        var enrtyAdmin = usersRepo.findById(body.get("admin"));
+        var enrtyUser = usersRepo.findById(body.get("user"));
+        if(enrtyAdmin.isEmpty()){
+            return "Wrong admin id";
+        }
+        if(enrtyUser.isEmpty()){
+            return "Wrong user id";
+        }
+        User admin = enrtyAdmin.get();
+        User user = enrtyUser.get();
+        if(!admin.getIs_admin()){
+            return "Admin is not admin";
+        }
+        user.setIs_admin(false);
+        usersRepo.save(user);
+        return "Admin removed successfully";
+    }
```

Comment:
Should be given a better name. The name "removeAdmin" may falsly indicate that the function deletes admins.